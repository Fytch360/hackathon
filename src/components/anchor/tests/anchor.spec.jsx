import React from 'react'
import { shallow } from 'enzyme'

import { Anchor, AnchorLink } from '..'

describe('<Anchor />', () => {

    test('is available', () => {
        expect(Anchor).toBeDefined()
    })

    test('snapshot', () => {
        const wrapper = shallow(
            <Anchor>
                <AnchorLink title="0" >{'0'}</AnchorLink>
                <AnchorLink title="1" >{'1'}</AnchorLink>
            </Anchor>
        )

        expect(wrapper).toMatchSnapshot()
    })

    test('Throwing size and colorScheme for tabs', () => {
        const inputProps1 = {
            size: 'sm',
            colorScheme: 'pink5',
        }

        const wrapper1 = shallow(
            <Anchor {...inputProps1}>
                <AnchorLink title="0" />
                <AnchorLink title="1" />
                <AnchorLink title="2" />
            </Anchor>
        )

        wrapper1.find(AnchorLink).forEach((link) => {
            expect(link.props()).toMatchObject(inputProps1)
        })

        const inputProps2 = {
            size: 'lg',
            colorScheme: 'orange5',
        }

        const wrapper2 = shallow(
            <Anchor {...inputProps2} >
                <AnchorLink title="0" />
                <AnchorLink title="1" />
                <AnchorLink title="2" />
            </Anchor>
        )
        
        wrapper2.find(AnchorLink).forEach((link) => {
            expect(link.props()).toMatchObject(inputProps2)
        })
    })

    test('With inintial value should throw forceOpened prop for correct AnchorLink', () => {
        const wrapper1 = shallow(
            <Anchor initialValue="1">
                <AnchorLink title="0" />
                <AnchorLink title="1" />
                <AnchorLink title="2" />
            </Anchor>
        )

        wrapper1.find(AnchorLink).forEach((link, i) => {
            if (i === 1) {
                expect(link.props().forceOpened).toBe(true)
            } else {
                expect(link.props().forceOpened).toBe(false)
            }
        })
        
        const wrapper2 = shallow(
            <Anchor initialValue="2">
                <AnchorLink title="0" />
                <AnchorLink title="1" />
                <AnchorLink title="2" />
            </Anchor>
        )

        wrapper2.find(AnchorLink).forEach((link, i) => {
            if (i === 2) {
                expect(link.props().forceOpened).toBe(true)
            } else {
                expect(link.props().forceOpened).toBe(false)
            }
        })
    })
})
