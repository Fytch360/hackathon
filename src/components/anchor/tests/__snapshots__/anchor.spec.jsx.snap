// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> snapshot 1`] = `
<TabsStyled>
  <Header
    parentId="anchor-link-wrapper-1"
  >
    <div>
      <AnchorLink
        forceOpened={true}
        key=".0"
        onChange={[Function]}
        parentId="anchor-link-wrapper-1"
        size="lg"
        title="0"
      >
        0
      </AnchorLink>
      <AnchorLink
        forceOpened={false}
        key=".1"
        onChange={[Function]}
        parentId="anchor-link-wrapper-1"
        size="lg"
        title="1"
      >
        1
      </AnchorLink>
    </div>
  </Header>
  <TabsContentSectionStyled>
    <div
      id="anchor-link-id-0"
      key="0-0"
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex="-1"
    >
      0
    </div>
    <div
      id="anchor-link-id-1"
      key="1-1"
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex="-1"
    >
      1
    </div>
  </TabsContentSectionStyled>
</TabsStyled>
`;
